examples/AspNetCoreExample/Dockerfile

```diff
@@ -1,12 +1,22 @@
-#FROM mcr.microsoft.com/dotnet/core/sdk:3.1.406 AS build
+#See https://aka.ms/containerfastmode to understand how Visual Studio uses this Dockerfile to build your images for faster debugging.
+
+FROM mcr.microsoft.com/dotnet/aspnet:5.0 AS base
+WORKDIR /app
+EXPOSE 80
+
 FROM mcr.microsoft.com/dotnet/sdk:5.0 AS build
 WORKDIR /src
+COPY ["examples/AspNetCoreExample/AspNetCoreExample.csproj", "examples/AspNetCoreExample/"]
+COPY ["src/prometheus-net.DotNetRuntime/prometheus-net.DotNetRuntime.csproj", "src/prometheus-net.DotNetRuntime/"]
+RUN dotnet restore "examples/AspNetCoreExample/AspNetCoreExample.csproj"
 COPY . .
-RUN dotnet publish "examples/AspNetCoreExample" -c Release -o /app
+WORKDIR "/src/examples/AspNetCoreExample"
+RUN dotnet build "AspNetCoreExample.csproj" -c Release -o /app/build
```

Comment:
There should be no need for an explicit `restore` or `build`- `publish` does all these things implicitly.